service/controller/clusterapi/v18/resources/certconfig/current.go

```diff
@@ -9,65 +9,49 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/giantswarm/cluster-operator/pkg/label"
-	"github.com/giantswarm/cluster-operator/pkg/v18/key"
+	"github.com/giantswarm/cluster-operator/service/controller/clusterapi/v18/key"
 )
 
 // GetCurrentState takes observed custom object as an input and based on that
 // information looks for current state of cluster certconfigs and returns them.
 // Return value is of type []*v1alpha1.CertConfig.
 func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
-	clusterGuestConfig, err := r.toClusterGuestConfigFunc(obj)
+	cr, err := key.ToCluster(obj)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
-	objectMeta, err := r.toClusterObjectMetaFunc(obj)
-	if err != nil {
-		return nil, microerror.Mask(err)
-	}
-
-	r.logger.LogCtx(ctx, "level", "debug", "message", "looking for a list of certconfigs in the Kubernetes API")
-
 	var certConfigs []*v1alpha1.CertConfig
 	{
-		labelSelector := &metav1.LabelSelector{}
-		labelSelector = metav1.AddLabelToSelector(labelSelector, label.Cluster, key.ClusterID(clusterGuestConfig))
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding certconfigs in namespace %#q", cr.Namespace))
 
-		selector, err := metav1.LabelSelectorAsSelector(labelSelector)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
-
-		listOptions := metav1.ListOptions{
-			LabelSelector: selector.String(),
+		o := metav1.ListOptions{
+			Continue:      "",
+			LabelSelector: fmt.Sprintf("%s=%s", label.Cluster, key.ClusterID(&cr)),
 			Limit:         listCertConfigLimit,
 		}
 
-		continueToken := ""
-
 		for {
```

Comment:
Forgive my ignorance, but do we need an infinite loop in here for some reasons?

Reply:
This is the implementation for paging of k8s api list results. We apparently want to limit the list of certconfigs we process in a row to 25, which we should actually never hit because we have less than 10 certconfigs per TC and the listing has a label selector option. I did not make this up though and only refactored it a bit. Do you think we should just drop that?

Reply:
So far we have it for a reason I'm okay with it :) Thanks for the explanation.